### PR TITLE
evm t8n tool to use ExecuteBlockEphemerally api

### PR DESCRIPTION
**Pull request by sudeepdino008**
- get evm t8n to use ExecuteBlockEphemerally
- unit tests to ensure that the outputs are same as corresponding geth's evm tool
- align structs used by the evm tool to match to the current status of geth's evm tool (for the purpose of updating to 
   capture new information and to achieve parity)

### CODE REVIEW

**Comment by AlexeyAkhunov** [comment]
Thank you! Can you please remove function `rlpHash` in the file `cmd/evm/internal/t8ntool/execution.go` to fix lint error

**Comment by sudeepdino008** [comment]
> Thank you! Can you please remove function `rlpHash` in the file `cmd/evm/internal/t8ntool/execution.go` to fix lint error

Done!

**Comment by AlexeyAkhunov** [comment]
Great work! Will need to give it a bit more review, especially `core/blockchain.go` and then merge. Thank you!

File: core/blockchain.go [hunk 1]
```diff
@@ -194,16 +222,21 @@ func ExecuteBlockEphemerallyForBSC(
 		newBlock = block
 	}
 
+	var bloom types.Bloom
+
+	receiptSha := types.DeriveSha(receipts)
```
Comment (AlexeyAkhunov) [inline core/blockchain.go:227]: Is this not the same as `newBlock.ReceiptHash()` ? I am just wondering if it gets calculated twice

Comment (sudeepdino008) [reply core/blockchain.go:227]: when `!vmConfig.ReadOnly` above, yes it's calculated twice. Refactored to remove that.

File: core/blockchain.go [hunk 2]
```diff
@@ -176,35 +204,76 @@ func ExecuteBlockEphemerallyForBSC(
...
 
-	return receipts, nil
+	var logs []*types.Log
+	for _, receipt := range receipts {
+		logs = append(logs, receipt.Logs...)
+	}
+
+	blockLogs := ibs.Logs()
+	var stateSyncReceipt *types.ReceiptForStorage
+	if len(blockLogs) > 0 {
```
Comment (AlexeyAkhunov) [inline core/blockchain.go:214]: Looks like copy-paste from the other function, `ExecuteBlockEphemerally`. Is this actually required for BSC? I think this is only beed for Bor consensus

Comment (AlexeyAkhunov) [reply core/blockchain.go:214]: Ok, I will check it out once it is removed

**Comment by AlexeyAkhunov** [comment]
Looks like merging this has broken some integration tests, will need to dig into this, but this is what I see for example:

```
go test ./tests --run TestBlockchain/TransitionTests/bcHomesteadToDao --tags integration

[EROR] [07-02|22:16:20.823] [4/16 Bodies] Uncle verification failed  number=7 hash=0x418435e9096986a6ba2a625e2a8610f74f1e16a530c6948c3d508a1bc071182d err="bad DAO pro-fork extra-data"
[EROR] [07-02|22:16:20.826] [4/16 Bodies] Uncle verification failed  number=8 hash=0xa3d7bebffb7e945595c9af5cee45faa8e0e353f28935fbffd0dc085f22e739a1 err="bad DAO pro-fork extra-data"
[EROR] [07-02|22:16:20.831] [4/16 Bodies] Uncle verification failed  number=10 hash=0x7313d768586c6108d211f5a007b6a43f5095e0e1147f96484af1aece5a2bb87b err="bad DAO pro-fork extra-data"
[EROR] [07-02|22:16:20.841] [4/16 Bodies] Uncle verification failed  number=15 hash=0x9d9339f527be0febb1713b38c88516010582b6bf372813bdd1aa7ddb20dacc99 err="bad DAO pro-fork extra-data"
[EROR] [07-02|22:16:20.843] [4/16 Bodies] Uncle verification failed  number=16 hash=0x58e70f91c436b2224cfafcc0c40eb0363980bbe69eb59e3389a2aa98605d1fb9 err="bad DAO pro-fork extra-data"
--- FAIL: TestBlockchain (0.01s)
    --- FAIL: TestBlockchain/TransitionTests/bcHomesteadToDao/DaoTransactions.json (0.20s)
        block_test.go:43: block #5 insertion into chain failed: did not import block 5 db1b027624fd6e966feef9445e4a50982e89390c7f9693c0e7f73cadcad03086
FAIL
FAIL	github.com/ledgerwatch/erigon/tests	0.735s
```

**Comment by AlexeyAkhunov** [comment]
I have asked @awskii to re-work and reapply this change (it is very useful, thank you for making it @sudeepdino008 ), and to make sure tests pass and there is no de-optimisation. Please feel free to connect

**Comment by sudeepdino008** [comment]
thanks @AlexeyAkhunov. 
My bad, didn't realize there were integration tests in there. Great that it got caught this early on.

**Comment by sudeepdino008** [comment]
@AlexeyAkhunov it's resolved https://github.com/ledgerwatch/erigon/pull/4642

**Pull request merged**